### PR TITLE
added check for fe be connection

### PR DESCRIPTION
**Pull request by NikitaPoly**

### CODE REVIEW

File: src/mantarray_desktop_app/main_process/queue_container.py [hunk 1]
```diff
@@ -41,6 +41,8 @@ def __init__(self) -> None:
         self.from_server: queue.Queue[Dict[str, Any]] = queue.Queue()
         self.to_server: LightQueue = LightQueue()
 
+        self.from_websocket: queue.Queue[Dict[str, Any]] = queue.Queue()
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/main_process/queue_container.py:44]: This needs to be a `LightQueue` like the other queue going to the websockets. We should probably also rename these queues now:
```
from_flask
to_websocket
from_websocket
```

File: tests/fixtures_process_monitor.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ def fixture_test_monitor():
     def _foo(process_manager):
         svd = process_manager.values_to_share_to_server
         svd["system_status"] = SERVER_INITIALIZING_STATE
+        svd["fe_be_connection_success"] = True
```
Comment (tannermpeterson) [inline tests/fixtures_process_monitor.py:15]: This fixture sets up the bare minimum in the shared_values_dict only, which is `system_status = SERVER_INITIALIZING_STATE`, so this key should not be set here

File: tests/main/test_main.py [hunk 3]
```diff
@@ -51,8 +51,9 @@ def test_set_up_socketio_handlers__sets_up_socketio_events_correctly(mocker, fsi
     mocked_start_bg_task = mocker.patch.object(main.socketio, "start_background_task", autospec=True)
 
     test_queue = TestingQueue()
+    test_queue_two = TestingQueue()
```
Comment (tannermpeterson) [inline tests/main/test_main.py:54]: This test should use better names now that there are two queues

File: CHANGELOG.rst [hunk 4]
```diff
@@ -1,5 +1,11 @@
 Changelog for Mantarray Desktop App
 ===================================
+1.0.7 (unreleased)
+------------------
+
+Added:
```
Comment (tannermpeterson) [inline CHANGELOG.rst:6]: The format of this is incorrect. Please see the subheadings for other versions to how it should be done

File: src/mantarray_desktop_app/main.py [hunk 5]
```diff
@@ -68,7 +68,9 @@ def get_server_port_number() -> int:
     return server_manager.get_port_number()
 
 
-def _set_up_socketio_handlers(ws_queue: LightQueue) -> Callable[[], None]:
+def _set_up_socketio_handlers(
+    ws_queue: LightQueue, to_pm_queue: queue.Queue[Dict[str, Any]]
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/main.py:72]: Like I mentioned earlier, the naming convention is always `to/from main` never `to/from pm`, so we'll need to rename this queue. It also makes sense to rename the other queue to `to_ws_queue` now so it's very clear what each queue is for

File: src/mantarray_desktop_app/main.py [hunk 6]
```diff
@@ -243,6 +246,7 @@ def main(command_line_args: List[str], object_access_for_testing: Optional[Dict[
 
         shared_values_dict = SharedValues()
 
+        shared_values_dict["fe_be_connection_success"] = False
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/main.py:249]: Should set this value right before `_set_up_socketio_handlers` is called. Also the name of this key should be more general, like `"websocket_connection_made"`

File: src/mantarray_desktop_app/main_process/process_monitor.py [hunk 7]
```diff
@@ -868,3 +872,13 @@ def _hard_stop_and_join_processes_and_log_leftovers(
...
+
+    def _check_and_handle_websocket_to_main_queue(self) -> None:
+        process_manager = self._process_manager
+        to_main_queue = process_manager.queue_container.from_websocket
+        try:
+            communication = to_main_queue.get(timeout=SECONDS_TO_WAIT_WHEN_POLLING_QUEUES)
+        except queue.Empty:
+            return
+        if communication["communication_type"] == "connection_success":
+            self._values_to_share_to_server["fe_be_connection_success"] = True
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/main_process/process_monitor.py:884]: Need to unit test this

File: src/mantarray_desktop_app/main_process/process_monitor.py [hunk 8]
```diff
@@ -786,7 +787,10 @@ def _commands_for_each_run_iteration(self) -> None:
     def _check_subprocess_start_up_statuses(self) -> None:
         process_manager = self._process_manager
         shared_values_dict = self._values_to_share_to_server
-        if process_manager.are_subprocess_start_ups_complete():
+        if (
+            process_manager.are_subprocess_start_ups_complete()
+            and shared_values_dict["fe_be_connection_success"]
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/main_process/process_monitor.py:792]: Need to update unit tests for this

File: tests/main/test_launch.py [hunk 9]
```diff
@@ -484,7 +484,8 @@ def test_main__full_launch_script_runs_as_expected(fully_running_app_from_main_e
 
     # assert socketio was set up correctly
     ws_queue = app_info["object_access_inside_main"]["process_manager"].queue_container.to_server
-    mocked_set_up.assert_called_once_with(ws_queue)
+    to_pm_queue = app_info["object_access_inside_main"]["process_manager"].queue_container.from_websocket
```
Comment (tannermpeterson) [inline tests/main/test_launch.py:487]: When grabbing queues from the queue container, you should try to use the same name so its' easier to follow. I know the line above doesn't do this either, so it would be good to fix that if you come across it anywhere

**Review by tannermpeterson — CHANGES_REQUESTED** [review]

File: src/mantarray_desktop_app/main_process/process_monitor.py [hunk 10]
```diff
@@ -91,15 +91,15 @@ class MantarrayProcessesMonitor(InfiniteThread):
 
     def __init__(
         self,
-        values_to_share_to_server: SharedValues,
+        values_to_share_to_websocket: SharedValues,
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/main_process/process_monitor.py:94]: I think we can keep the name of this the same

File: src/mantarray_desktop_app/main_process/process_monitor.py [hunk 11]
```diff
@@ -799,14 +801,17 @@ def _commands_for_each_run_iteration(self) -> None:
 
     def _check_subprocess_start_up_statuses(self) -> None:
         process_manager = self._process_manager
-        shared_values_dict = self._values_to_share_to_server
-        if process_manager.are_subprocess_start_ups_complete():
+        shared_values_dict = self._values_to_share_to_websocket
+        if (
+            process_manager.are_subprocess_start_ups_complete()
+            and shared_values_dict["websocket_connection_made"]
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/main_process/process_monitor.py:807]: 👍

File: tests/fixtures.py [hunk 12]
```diff
@@ -147,7 +147,7 @@ def _foo(beta_2_mode=False, create_processes=True, use_testing_queues=False):
             recording_dir = os.path.join(tmp_dir, "recordings")
             mag_analysis_output_dir = os.path.join(tmp_dir, "time_force_data")
             manager = MantarrayProcessesManager(
-                values_to_share_to_server={
+                values_to_share_to_websocket={
```
Comment (tannermpeterson) [inline tests/fixtures.py:150]: This should also stay `values_to_share_to_server`

File: tests/fixtures_server.py [hunk 13]
```diff
@@ -170,12 +170,12 @@ def twitch_metrics_handler(data):
     def stimulation_handler(data):
         msg_list_container["stimulation_data"].append(data)
 
-    def _connect_client_to_server():
+    def _connect_client_to_websocket():
```
Comment (tannermpeterson) [inline tests/fixtures_server.py:173]: Might want to make this `_connect_client_to_ws_server` to clear up ambiguity

File: tests/process_monitor/test_comm_from_server.py [hunk 14]
```diff
@@ -957,3 +957,27 @@ def test_MantarrayProcessesMonitor__processes_firmware_update_confirmation_comma
...
     # make sure system status was not updated
     assert shared_values_dict["system_status"] == UPDATES_NEEDED_STATE
+
+
+def test_MantarrayProcessesMonitor___check_and_handle_websocket_to_main_queue_sets_flag_when_websocket_connection_success(
+    test_process_manager_creator, test_monitor, patch_subprocess_joins, mocker
+):
+    test_process_manager = test_process_manager_creator(use_testing_queues=True)
+    monitor_thread, shared_values_dict, *_ = test_monitor(test_process_manager)
+    to_main_queue = test_process_manager.queue_container.from_websocket
```
Comment (tannermpeterson) [inline tests/process_monitor/test_comm_from_server.py:967]: Should call this variable `from_websocket_queue` since there are multiple queues going to main

File: tests/process_monitor/test_comm_from_server.py [hunk 15]
```diff
@@ -957,3 +957,27 @@ def test_MantarrayProcessesMonitor__processes_firmware_update_confirmation_comma
...
+
+    shared_values_dict["websocket_connection_made"] = False
+
+    communication = {"communication_type": "connection_success"}
+
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(communication, to_main_queue)
+    invoke_process_run_and_check_errors(monitor_thread)
+
+    mocked_successful_connection = mocker.patch.object(
```
Comment (tannermpeterson) [inline tests/process_monitor/test_comm_from_server.py:976]: This mocking is unnecessary here. The mocked function will never be run since `invoke_process_run_and_check_errors(monitor_thread)` is called before this function is even mocked. Also, the `se` function below can be removed as well

File: tests/process_monitor/test_process_monitor.py [hunk 16]
```diff
@@ -372,7 +372,7 @@ def test_MantarrayProcessesMonitor__handles_instrument_related_errors_from_instr
     monitor_thread, *_ = test_monitor(test_process_manager)
     ic_process = test_process_manager.instrument_comm_process
     ic_error_queue = test_process_manager.queue_container.instrument_comm_error
-    queue_to_server_ws = test_process_manager.queue_container.to_server
+    queue_to_websocket_ws = test_process_manager.queue_container.to_websocket
```
Comment (tannermpeterson) [inline tests/process_monitor/test_process_monitor.py:375]: This can just be `queue_to_websocket`

File: tests/process_monitor/test_process_monitor.py [hunk 17]
```diff
@@ -627,7 +627,7 @@ def test_MantarrayProcessesMonitor__correctly_sets_system_status_to_live_view_ac
     assert shared_values_dict["system_status"] == RECORDING_STATE
 
 
-def test_MantarrayProcessesMonitor__sets_system_status_to_server_ready_after_subprocesses_finish_start_up(
+def test_MantarrayProcessesMonitor__sets_system_status_to_websocket_ready_after_subprocesses_finish_start_up(
```
Comment (tannermpeterson) [inline tests/process_monitor/test_process_monitor.py:630]: This test name should be test_MantarrayProcessesMonitor__sets_system_status_to_server_ready_after_subprocesses_finish_start_up__and_websocket_connection_is_made

File: tests/test_process_manager.py [hunk 18]
```diff
@@ -365,12 +365,12 @@ def test_MantarrayProcessesManager__passes_file_directory_to_FileWriter():
     assert manager.file_writer_process.get_file_directory() == test_dir
 
 
-def test_MantarrayProcessesManager__passes_shared_values_dict_to_server():
+def test_MantarrayProcessesManager__passes_shared_values_dict_to_websocket():
```
Comment (tannermpeterson) [inline tests/test_process_manager.py:368]: The name of this test shouldn't change

File: tests/test_process_manager.py [hunk 19]
```diff
@@ -479,7 +479,7 @@ def test_MantarrayProcessesManager__are_processes_stopped__waits_correct_amount_
     assert mocked_da_is_stopped.call_count == 1
 
 
-def test_MantarrayProcessesManager__create_processes__passes_port_value_from_dictionary_to_server_manager(
+def test_MantarrayProcessesManager__create_processes__passes_port_value_from_dictionary_to_websocket_manager(
```
Comment (tannermpeterson) [inline tests/test_process_manager.py:482]: The name of this test shouldn't change

File: tests/test_process_manager.py [hunk 20]
```diff
@@ -609,7 +609,7 @@ def test_MantarrayProcessesManager__creates_mc_comm_instead_of_ok_comm_when_beta
     assert isinstance(mc_comm_process, McCommunicationProcess) is True
 
 
-def test_MantarrayProcessesManager_shutdown_server__shutsdown_server_and_returns_remaining_items_in_queue_to_server(
+def test_MantarrayProcessesManager_shutdown_server__shutsdown_server_and_returns_remaining_items_in_queue_to_websocket(
```
Comment (tannermpeterson) [inline tests/test_process_manager.py:612]: The name of this test shouldn't change

File: src/mantarray_desktop_app/utils/system.py [hunk 21]
```diff
@@ -37,7 +37,7 @@ def wait_for_subprocesses_to_start() -> None:
     start = time.perf_counter()
     elapsed_time = 0.0
     response: Optional[Response]
-    while elapsed_time < 20:
+    while elapsed_time < 50:
```
Comment (tannermpeterson) [inline src/mantarray_desktop_app/utils/system.py:40]: Changing this value won't help fix the tests, we should keep it at 20

File: tests/main/test_launch.py [hunk 22]
```diff
@@ -293,7 +293,10 @@ def test_main_entrypoint__correctly_assigns_shared_values_dictionary_to_process_
     object_access_dict = app_info["object_access_inside_main"]
     shared_values_dict = object_access_dict["values_to_share_to_server"]
     test_process_monitor = object_access_dict["process_monitor"]
-    assert test_process_monitor._values_to_share_to_server is shared_values_dict
+    assert (
+        test_process_monitor._values_to_share_to_server  # pylint: disable=protected-access
```
Comment (tannermpeterson) [inline tests/main/test_launch.py:297]: I don't think we're using pylint anymore, is there any reason for this comment?

Comment (NikitaPoly) [reply tests/main/test_launch.py:297]: I don't remember adding this. I think it might be a product of the pre-commits I ran. I will remove it

File: tests/process_monitor/test_process_monitor.py [hunk 23]
```diff
@@ -296,7 +296,7 @@ def test_MantarrayProcessesMonitor__handled_completed_mag_analysis_command_corre
     invoke_process_run_and_check_errors(monitor_thread)
     assert is_queue_eventually_empty(data_analyzer_to_main) is True
 
-    ws_message = queue_to_server_ws.get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)
+    ws_message = queue_to_websocket_ws.get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)
```
Comment (tannermpeterson) [inline tests/process_monitor/test_process_monitor.py:299]: Be careful when doing mass find + replace, this name should just be `queue_to_websocket`. I think I saw this happen in some other files as well

File: tests/main/test_integration.py [hunk 24]
```diff
@@ -113,7 +114,7 @@
 ]
 LIVE_VIEW_ACTIVE_WAIT_TIME = 150
 CALIBRATED_WAIT_TIME = 20
-STOP_MANAGED_ACQUISITION_WAIT_TIME = 40
+STOP_MANAGED_ACQUISITION_WAIT_TIME = 200
```
Comment (tannermpeterson) [inline tests/main/test_integration.py:117]: None of these timeout values should change

File: tests/main/test_integration.py [hunk 25]
```diff
@@ -249,6 +251,7 @@ def test_full_datapath_and_recorded_files_in_beta_1_mode(
         # Tanner (7/14/21): Beta 1 data packets are sent once per second, so there should be at least one data packet for every second needed to run analysis, but sometimes the final data packet doesn't get sent in time
         assert len(msg_list_container["waveform_data"]) >= MIN_NUM_SECONDS_NEEDED_FOR_ANALYSIS - 1
         confirm_queue_is_eventually_empty(da_out, timeout_seconds=5)
+        sio.disconnect()
```
Comment (tannermpeterson) [inline tests/main/test_integration.py:254]: Why was this added?

**Review by tannermpeterson — APPROVED** [review]
